lib/websocket/receiver.js

```diff
@@ -254,6 +233,43 @@ class ByteParser extends Writable {
     }
   }
 
+  /**
+   * Take n bytes from the buffered Buffers
+   * @param {number} n
+   * @returns {Buffer|null}
+   */
+  consume (n) {
+    if (n > this.#byteOffset) {
+      return null
+    } else if (n === 0) {
+      return Buffer.allocUnsafe(0)
```

Comment:
Consider adding an empty buffer constant so that you can reuse it every time you need a zero length buffer without calling the constructor every time.